### PR TITLE
V3.21.0

### PR DESCRIPTION
**Pull request by emthompson-usgs**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 {% set name = "openquake.engine" %}
-{% set version = "3.19.0" %}
+{% set version = "3.21.0" %}
+{% set python_min = "3.7" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: The minimum numba required is not installable on Python 3.7.

```suggestion
{% set python_min = "3.8" %}
```

Comment (emthompson-usgs) [reply recipe/meta.yaml]: Thanks. I will fix this.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,14 +1,15 @@
 {% set name = "openquake.engine" %}
-{% set version = "3.19.0" %}
+{% set version = "3.21.0" %}
+{% set python_min = "3.8" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I changed to 3.8 b/c that is the lowest one where the min numba required here is resolved. However, it seems that openquake.engine requires Python 3.9 as a minimum. See https://github.com/gem/oq-engine/blob/b75b934f1b3400f1bd77c3488c817220e73fd3fd/setup.py#L24

```suggestion
{% set python_min = "3.9" %}
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,12 +24,13 @@ build:
...
+    - setuptools >=69.0
     - pip
   run:
+    - python >={{ python_min }}
     - alpha-shapes >=1.1.0
-    - python >=3.8
-    - setuptools
+    - setuptools >=69.0
     - h5py >=2.10
     - numpy >=1.20
```
Comment (ocefpaf) [inline recipe/meta.yaml]: This is not in the metadata and it is an easy fix upstream. You will need to import the rankwarning from the new excpetion name espace in numpy 2.0. For now, pinning it to <2 would do the trick.

```suggestion
    - numpy >=1.20,<2
```

**Pull request merged**